test/src/test/java/hudson/model/QueueTest.java

```diff
@@ -963,10 +972,9 @@ public void save(){
     }
 
     @Test public void waitForStartAndCancelBeforeStart() throws Exception {
-        final OneShotEvent ev = new OneShotEvent();
         FreeStyleProject p = r.createFreeStyleProject();
 
-        QueueTaskFuture<FreeStyleBuild> f = p.scheduleBuild2(10);
+        QueueTaskFuture<FreeStyleBuild> f = p.scheduleBuild2(30);
```

Comment:
This was effectively giving the test up to 10 seconds to pass. A bit bold on slow systems! This gives the test up to 30 seconds to pass which should allow for running the test successfully on slow systems (such as systems that are intentionally CPU bound).